a8c-files.php

```diff
@@ -511,6 +514,88 @@ public function update_attachment_meta() {
 
 		update_option( self::OPT_NEXT_FILESIZE_INDEX, $start_index, false );
 	}
+
+	// phpcs:disable Generic.CodeAnalysis.UnusedFunctionParameter.FoundAfterLastUsed
+	/**
+	 * Adds 'correct' w & h values to img tags in core/image blocks
+	 *
+	 * @param string $block_content The block content.
+	 * @param array  $block The full block, including name and attributes.
+	 * @param object $instance The block instance.
+	 * @return string Modified block content.
+	 */
+	public function fix_img_block_sizes( $block_content, $block, $instance ) {
+		// Don't fire in wp-admin, image blocks are fine there
+		if ( is_admin() ) {
+			return $block_content;
+		}
+
+		// If no HTML API, bail
+		if ( ! class_exists( 'WP_HTML_Tag_Processor' ) ) {
+			return $block_content;
+		}
+
+		// Check img tag
+		$img_tag_processor = new WP_HTML_Tag_Processor( $block_content );
+		if ( ! $img_tag_processor->next_tag( 'img' ) ) {
+			// No img found, bail
+			return $block_content;
+		}
+
+		// Get the image ID class from the img tag
+		$img_class = $img_tag_processor->get_attribute( 'class' );
+		if ( ! $img_class || ! preg_match( '/wp-image-(\d+)/', $img_class, $image_matches ) ) {
+			// If we can't find wp-image-XXX, bail
+			return $block_content;
+		}
+		$image_id = $image_matches[1];
+
+		// Check figure tag
+		$figure_tag_processor = new WP_HTML_Tag_Processor( $block_content );
+		if ( ! $figure_tag_processor->next_tag( 'figure' ) ) {
+			// No figure found, bail
+			return $block_content;
+		}
+
+		// Get the size name class from the figure tag
+		$figure_class = $figure_tag_processor->get_attribute( 'class' );
+		if ( ! $figure_class || ! preg_match( '/size-([^\s"]+)/', $figure_class, $size_matches ) ) {
+			// If we can't find size-xxx, bail
+			return $block_content;
+		}
+		$size_name = $size_matches[1];
+
+		// If the size is full, bail
+		if ( 'full' === $size_name ) {
+			return $block_content;
+		}
+
+		// Get meta for the attachment sizes
+		$metadata = wp_get_attachment_metadata( (int) $image_id );
+
+		if ( ! $metadata || empty( $metadata['sizes'][ $size_name ] ) ) {
+			// If no size meta is available for this size, bail
+			return $block_content;
+		}
+
+		// Get values ready
+		$width                = $metadata['sizes'][ $size_name ]['width'];
+		$height               = $metadata['sizes'][ $size_name ]['height'];
+		$update_tag_processor = new WP_HTML_Tag_Processor( $block_content );
+
+		// Set width/height for img
+		if ( $update_tag_processor->next_tag( 'img' ) ) {
+			$update_tag_processor->set_attribute( 'width', esc_attr( $width ) );
+			$update_tag_processor->set_attribute( 'height', esc_attr( $height ) );
```

Comment:
```suggestion
			$update_tag_processor->set_attribute( 'width', $width );
			$update_tag_processor->set_attribute( 'height', $height );
```

See https://developer.wordpress.org/reference/classes/wp_html_tag_processor/set_attribute/:
> For string attributes, the value is escaped using the `esc_attr` function.

See https://github.com/WordPress/wordpress-develop/blob/50fb4086b7afbfa012c5d1f2eeff79b1bae3b00e/src/wp-includes/html-api/class-wp-html-tag-processor.php#L3875